qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/FormSteps.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.ClinicianFacingPatientDashboardPage;
+import org.openmrs.contrib.qaframework.page.HomePage;
+import org.openmrs.contrib.qaframework.page.HtmlFormsPage;
+import org.openmrs.contrib.qaframework.page.ManageFormsPage;
+import org.openmrs.contrib.qaframework.page.ManageHtmlFormsPage;
+import org.openqa.selenium.By;
+
+public class FormSteps extends Steps {
+	private static final String NAME = "newFormTest1";
+	private static final String DESCRIPTION = "description of new form";
+	private static final String VERSION = "1.2";
+	private HomePage homePage;
+	private ManageFormsPage manageForm;
+	private ManageHtmlFormsPage manageHtmlFormsPage;
+	private HtmlFormsPage htmlFormsPage;
+	private ClinicianFacingPatientDashboardPage patientDashboardPage;
```

Comment:
do you need to declare ClinicianFacingPatientDashboardPage as a variable?

Reply:
We are using its variable in our assertions. Similar to homepage. we are using homePage in the before method

Reply:
why not take advantage of inheritance?

Reply:
Are you meaning to use inheritance when testBase doesn't have such class. We simply need to instantiate these classes in before hook and use that advantage of it

Reply:
FormSteps is a child class to Steps. By [inheritance](https://www.javatpoint.com/inheritance-in-java) this makes the child class to access all the properties and behaviors of a parent object(Steps).

By declaring again the variable in the child class creates code redundancy. 

@sherrif10 why not just call the variable name([dashboardPage](https://github.com/openmrs/openmrs-contrib-qaframework/blob/586bd60662dd39fb0bb489e6e8d0b93fae0bdf25/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/Steps.java#L37)) within FormSteps class?

Reply:
Resolved thanks.